pinot-spi/src/main/java/org/apache/pinot/spi/metrics/PinotMetricName.java

```diff
@@ -23,5 +23,18 @@
  */
 public interface PinotMetricName {
 
+  /**
+   * Returns the actual metric name.
+   */
   Object getMetricName();
+
+  /**
+   * Overrides the equals method.
+   */
+  boolean equals(Object obj);
```

Comment:
If you want to add `equals()` and `hashCode()` to the interface, I think that you need to add specific documentation on what the expected overridden implementation would be.

Reply:
More description added for that.